src/back/wgsl/writer.rs

```diff
@@ -973,10 +980,30 @@ impl<W: Write> Writer<W> {
             }
             // TODO: copy-paste from glsl-out
             Expression::AccessIndex { base, index } => {
-                self.write_expr(module, base, func_ctx)?;
-
                 let base_ty_res = &func_ctx.info[base].ty;
                 let mut resolved = base_ty_res.inner_with(&module.types);
+
+                let deref = match *resolved {
+                    TypeInner::Pointer { .. } => {
+                        match func_ctx.expressions[base] {
+                            // Variables have a pointer return type but shouldn't be dereferenced
+                            Expression::LocalVariable(_) | Expression::GlobalVariable(_)
+                            // Access chains might have pointer types but glsl considers them as references
```

Comment:
this mentions `glsl`, but we are in the WGSL backend, it shouldn't be influenced by GLSL